
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
      <link rel="icon" href="../../images/favicon_non_transparent.svg">
      <meta name="generator" content="mkdocs-1.3.1, mkdocs-material-8.4.2">
    
    
      
        <title>Lecture 4 - EEB430 Modelling in Ecology and Evolution</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.69437709.min.css">
      
        
        <link rel="stylesheet" href="../../assets/stylesheets/palette.cbb835fc.min.css">
        
      
      
    
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../stylesheets/extra.css">
    
    <script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
  </head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="green" data-md-color-accent="purple">
  
    
    
      <script>var palette=__md_get("__palette");if(palette&&"object"==typeof palette.color)for(var key of Object.keys(palette.color))document.body.setAttribute("data-md-color-"+key,palette.color[key])</script>
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#lecture-4-one-locus-selection" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../.." title="EEB430 Modelling in Ecology and Evolution" class="md-header__button md-logo" aria-label="EEB430 Modelling in Ecology and Evolution" data-md-component="logo">
      
  <img src="../../images/favicon_non_transparent.svg" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            EEB430 Modelling in Ecology and Evolution
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Lecture 4
            
          </span>
        </div>
      </div>
    </div>
    
      <form class="md-header__option" data-md-component="palette">
        
          
          
          <input class="md-option" data-md-color-media="" data-md-color-scheme="default" data-md-color-primary="green" data-md-color-accent="purple"  aria-hidden="true"  type="radio" name="__palette" id="__palette_1">
          
        
      </form>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" aria-label="Clear" tabindex="-1">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
            
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-tabs__inner md-grid">
    <ul class="md-tabs__list">
      
        
  
  


  <li class="md-tabs__item">
    <a href="../.." class="md-tabs__link">
      Overview
    </a>
  </li>

      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../../syllabus/general_info/" class="md-tabs__link">
        Syllabus
      </a>
    </li>
  

      
        
  
  
    
  


  
  
  
    <li class="md-tabs__item">
      <a href="../schedule/" class="md-tabs__link md-tabs__link--active">
        Lectures
      </a>
    </li>
  

      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../../labs/schedule/" class="md-tabs__link">
        Labs
      </a>
    </li>
  

      
        
  
  


  <li class="md-tabs__item">
    <a href="../../acknowledgements/" class="md-tabs__link">
      Acknowledgements
    </a>
  </li>

      
    </ul>
  </div>
</nav>
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

  


<nav class="md-nav md-nav--primary md-nav--lifted" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="EEB430 Modelling in Ecology and Evolution" class="md-nav__button md-logo" aria-label="EEB430 Modelling in Ecology and Evolution" data-md-component="logo">
      
  <img src="../../images/favicon_non_transparent.svg" alt="logo">

    </a>
    EEB430 Modelling in Ecology and Evolution
  </label>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../.." class="md-nav__link">
        Overview
      </a>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_2" type="checkbox" id="__nav_2" >
      
      
      
      
        <label class="md-nav__link" for="__nav_2">
          Syllabus
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Syllabus" data-md-level="1">
        <label class="md-nav__title" for="__nav_2">
          <span class="md-nav__icon md-icon"></span>
          Syllabus
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../syllabus/general_info/" class="md-nav__link">
        General info
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../syllabus/course_structure/" class="md-nav__link">
        Course structure
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../syllabus/textbook/" class="md-nav__link">
        Textbook
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../syllabus/when_and_where/" class="md-nav__link">
        When and where
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../syllabus/assignments/" class="md-nav__link">
        Assignments
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../syllabus/final_project/" class="md-nav__link">
        Final Project
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../syllabus/instructors/" class="md-nav__link">
        Instructors
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../syllabus/other_resources/" class="md-nav__link">
        Other resources
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
    
  
  
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_3" type="checkbox" id="__nav_3" checked>
      
      
      
      
        <label class="md-nav__link" for="__nav_3">
          Lectures
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Lectures" data-md-level="1">
        <label class="md-nav__title" for="__nav_3">
          <span class="md-nav__icon md-icon"></span>
          Lectures
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../schedule/" class="md-nav__link">
        Schedule
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../lecture-01/" class="md-nav__link">
        Lecture 1
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../lecture-02/" class="md-nav__link">
        Lecture 2
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../lecture-03/" class="md-nav__link">
        Lecture 3
      </a>
    </li>
  

            
          
            
              
  
  
    
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" data-md-toggle="toc" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          Lecture 4
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        Lecture 4
      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#lecture-overview" class="md-nav__link">
    Lecture overview
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#1-review-of-hardy-weinberg" class="md-nav__link">
    1. Review of Hardy-Weinberg
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#2-one-locus-haploid-selection-in-discrete-time" class="md-nav__link">
    2. One-locus haploid selection in discrete time
  </a>
  
    <nav class="md-nav" aria-label="2. One-locus haploid selection in discrete time">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#life-cycle-diagram" class="md-nav__link">
    Life-cycle diagram
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#deriving-the-equations" class="md-nav__link">
    Deriving the equations
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#simplifying" class="md-nav__link">
    Simplifying
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#3-one-locus-diploid-selection-in-discrete-time" class="md-nav__link">
    3. One-locus diploid selection in discrete time
  </a>
  
    <nav class="md-nav" aria-label="3. One-locus diploid selection in discrete time">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#life-cycle-diagram_1" class="md-nav__link">
    Life-cycle diagram
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#deriving-the-equations_1" class="md-nav__link">
    Deriving the equations
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#simplifying_1" class="md-nav__link">
    Simplifying
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#4-comparing-haploid-and-diploid-selection" class="md-nav__link">
    4. Comparing haploid and diploid selection
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#5-one-locus-haploid-selection-in-continuous-time" class="md-nav__link">
    5. One-locus haploid selection in continuous time
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#6-comparing-haploid-selection-in-discrete-and-continuous-time" class="md-nav__link">
    6. Comparing haploid selection in discrete and continuous time
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#7-the-models-weve-covered" class="md-nav__link">
    7. The models we've covered
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_4" type="checkbox" id="__nav_4" >
      
      
      
      
        <label class="md-nav__link" for="__nav_4">
          Labs
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Labs" data-md-level="1">
        <label class="md-nav__title" for="__nav_4">
          <span class="md-nav__icon md-icon"></span>
          Labs
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../labs/schedule/" class="md-nav__link">
        Schedule
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../acknowledgements/" class="md-nav__link">
        Acknowledgements
      </a>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#lecture-overview" class="md-nav__link">
    Lecture overview
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#1-review-of-hardy-weinberg" class="md-nav__link">
    1. Review of Hardy-Weinberg
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#2-one-locus-haploid-selection-in-discrete-time" class="md-nav__link">
    2. One-locus haploid selection in discrete time
  </a>
  
    <nav class="md-nav" aria-label="2. One-locus haploid selection in discrete time">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#life-cycle-diagram" class="md-nav__link">
    Life-cycle diagram
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#deriving-the-equations" class="md-nav__link">
    Deriving the equations
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#simplifying" class="md-nav__link">
    Simplifying
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#3-one-locus-diploid-selection-in-discrete-time" class="md-nav__link">
    3. One-locus diploid selection in discrete time
  </a>
  
    <nav class="md-nav" aria-label="3. One-locus diploid selection in discrete time">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#life-cycle-diagram_1" class="md-nav__link">
    Life-cycle diagram
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#deriving-the-equations_1" class="md-nav__link">
    Deriving the equations
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#simplifying_1" class="md-nav__link">
    Simplifying
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#4-comparing-haploid-and-diploid-selection" class="md-nav__link">
    4. Comparing haploid and diploid selection
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#5-one-locus-haploid-selection-in-continuous-time" class="md-nav__link">
    5. One-locus haploid selection in continuous time
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#6-comparing-haploid-selection-in-discrete-and-continuous-time" class="md-nav__link">
    6. Comparing haploid selection in discrete and continuous time
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#7-the-models-weve-covered" class="md-nav__link">
    7. The models we've covered
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content" data-md-component="content">
            <article class="md-content__inner md-typeset">
              
                


<script type="text/x-thebe-config">
  {
      requestKernel: true,
      mountActivateWidget: true,
      mountStatusWidget: true,
      binderOptions: {
      repo: "tomouellette/executable-cells",
      ref: "main",
      binderUrl: "https://gke.mybinder.org",
      },
  }
</script>
<script src="https://unpkg.com/thebe@latest/lib/index.js"></script>

<h1 id="lecture-4-one-locus-selection">Lecture 4: One-locus selection</h1>
<hr style="margin-bottom: 0em;">
<p><center></p>
<div class="inrow">
    Run notes interactively?
    <div style="float: left;" class="thebe-activate"></div>
    <div class="thebe-status"></div>
</div>
<p></center></p>
<hr style="margin-top: 0em;">

<h2 id="lecture-overview">Lecture overview</h2>
<ol>
<li><a href="#section1">Review of Hardy-Weinberg</a></li>
<li><a href="#section2">One-locus haploid selection in discrete time</a></li>
<li><a href="#section3">One-locus diploid selection in discrete time</a></li>
<li><a href="#section4">Comparing haploid and diploid selection</a></li>
<li><a href="#section5">One-locus haploid selection in continuous time</a></li>
<li><a href="#section6">Comparing haploid selection in discrete and continuous time</a></li>
<li><a href="#section7">The models we've covered</a></li>
</ol>
<p><span id='section1'></span></p>
<h2 id="1-review-of-hardy-weinberg">1. Review of Hardy-Weinberg</h2>
<hr>

<p>In the absence of selection, one round of random mating causes the frequency of diploid genotypes at a locus with two alleles, <span class="arithmatex">\(A\)</span> and <span class="arithmatex">\(a\)</span>, to become</p>
<ul>
<li><span class="arithmatex">\(AA\)</span>: <span class="arithmatex">\(p^2\)</span></li>
<li><span class="arithmatex">\(Aa\)</span>: <span class="arithmatex">\(2pq\)</span></li>
<li><span class="arithmatex">\(aa\)</span>: <span class="arithmatex">\(q^2\)</span></li>
</ul>
<p>where <span class="arithmatex">\(p\)</span> is the frequency of allele <span class="arithmatex">\(A\)</span> and <span class="arithmatex">\(q=1-p\)</span> is the frequency of allele <span class="arithmatex">\(a\)</span>. </p>
<p>A population with these diploid genotype frequencies is said to be in Hardy-Weinberg equilibrium. Furthermore, we showed that under this model the allele frequencies do not change, <span class="arithmatex">\(p' = p\)</span>. </p>
<p>But what if there <em>is</em> selection?</p>
<p><span id='section2'></span></p>
<h2 id="2-one-locus-haploid-selection-in-discrete-time">2. One-locus haploid selection in discrete time</h2>
<hr>

<p>We begin by examining a model where selection acts during the haploid phase of the life-cycle. This is a little simpler than diploid selection because there are only two haploid genotypes, <span class="arithmatex">\(A\)</span> and <span class="arithmatex">\(a\)</span>. It is also a very relevant model for species with long haploid phases (eg, some algae and fungi) or with strong competition during the haploid phase (eg, pollen competiting for ovules).</p>
<h3 id="life-cycle-diagram">Life-cycle diagram</h3>
<p>We first illustrate the structure of the model with a life-cycle diagram.</p>
<p><center></p>
<pre class="mermaid"><code>    graph LR;
    A((random&lt;br&gt; union)) --diploid--&gt; B((meiosis));
    B --haploid--&gt; C((selection));
    C --haploid--&gt; A;</code></pre>
<p></center></p>
<p><center><sup>Figure. Life-cycle diagram of haploid selection</sup></center></p>
<p>We will census the population at the beginning of the haploid phase (immediately after meiosis).</p>
<h3 id="deriving-the-equations">Deriving the equations</h3>
<p>Let</p>
<ul>
<li>
<p><span class="arithmatex">\(n_A(t) =\)</span> number of <span class="arithmatex">\(A\)</span> genotypes in generation <span class="arithmatex">\(t\)</span></p>
</li>
<li>
<p><span class="arithmatex">\(n_a(t) =\)</span> number of <span class="arithmatex">\(a\)</span> genotypes in generation <span class="arithmatex">\(t\)</span></p>
</li>
</ul>
<p>The frequency of <span class="arithmatex">\(A\)</span> is therefore <span class="arithmatex">\(p_t = \frac{n_a(t)}{n_A(t) + n_a(t)}\)</span>.</p>
<p>Now, let’s assume that not all haploid individuals survive to reproduce, and that the viability of an individual depends on its genotype:</p>
<ul>
<li>
<p><span class="arithmatex">\(v_A =\)</span> fraction of <span class="arithmatex">\(A\)</span> genotypes that survive</p>
</li>
<li>
<p><span class="arithmatex">\(v_a =\)</span> fraction of <span class="arithmatex">\(a\)</span> genotypes that survive</p>
</li>
</ul>
<p>After these deaths have occurred, there are <span class="arithmatex">\(v_A n_A(t)\)</span> <span class="arithmatex">\(A\)</span> genotypes and <span class="arithmatex">\(v_a n_a(t)\)</span> <span class="arithmatex">\(a\)</span> genotypes. The frequency of <span class="arithmatex">\(A\)</span> among these survivors is therefore
<span class="arithmatex">\(\frac{v_A n_A(t)}{v_A n_A(t) + v_a n_a(t)}\)</span>.</p>
<p>These survivors go on to unite at random, go through the diploid phase of the life-cycle without selection, and then undergo meiosis to produce the next generation of haploids. Because none of these processes change allele frequencies (as in our model of Hardy-Weinberg), the frequency of <span class="arithmatex">\(A\)</span> in the next generation is the frequency after death,</p>
<div class="arithmatex">\[
p_{t+1} = \frac{v_A n_A(t)}{v_A n_A(t) + v_a n_a(t)}.
\]</div>
<p>To make this a recursion equation we need to write <span class="arithmatex">\(p_{t+1}\)</span> in terms of <span class="arithmatex">\(p_t\)</span>, so that if we knew <span class="arithmatex">\(p\)</span> at some point in time we can recursively calculate it in all future times. To do this we divide both the numerator and denominator by the total number of individuals, <span class="arithmatex">\(n_A(t) + n_a(t)\)</span>, and simplify</p>
<div class="arithmatex">\[
\begin{aligned}
p_{t+1} 
&amp;= \frac{v_A n_A(t)}{v_A n_A(t) + v_a n_a(t)} \\
&amp;= \frac{v_A\frac{n_A(t)}{n_A(t) + n_a(t)}}{v_A\frac{n_A(t)}{n_A(t) + n_a(t)} + v_a\frac{n_a(t)}{n_A(t) + n_a(t)}}\\
&amp;= \frac{v_A p_t}{v_A p_t + v_a q_t}.
\end{aligned}
\]</div>
<p>This is now a recursion equation for the allele frequency in our model of haploid selection.</p>
<h3 id="simplifying">Simplifying</h3>
<p>Our current recursion is a function of two parameters, the absolute viabilities, <span class="arithmatex">\(v_A\)</span> and <span class="arithmatex">\(v_a\)</span>. Now notice that if we divide both the numerator and denominator by one of these viabilities, say <span class="arithmatex">\(v_a\)</span>, then we can reduce the recursion to a function of only one parameter, <span class="arithmatex">\(W_A = v_A/v_a\)</span>, the viability of <span class="arithmatex">\(A\)</span> <em>relative</em> to the viablity of <span class="arithmatex">\(a\)</span>,</p>
<div class="arithmatex">\[
\begin{aligned}
p_{t+1} &amp;= \frac{(v_A/v_a)p_t}{(v_A/v_a)p_t + (v_a/v_a)q_t}\\
&amp;= \frac{W_A p_t}{W_A p_t + q_t}.
\end{aligned}
\]</div>
<p>Let's plot the recursion to get a sense of the dynamics</p>
<pre data-executable="true" data-language="python">
import matplotlib.pyplot as plt

# calculate allele frequency over time with recursion
W_A = 1.1 #relative viability of allele A
p_now = 0.01 #initial allele frequency, p_0
ps = [] #list to hold allele frequencies over time
ts = range(100) #list of time steps
for t in ts: #for each time
    ps.append(p_now) #add current allele frequency to list
    p_now = W_A*p_now/(W_A*p_now+(1-p_now)) #update allele frequency with our recursion equation

# plot
plt.scatter(ts, ps) #plot the (t,p) pairs
plt.xlabel("generation, $t$") #label axes
plt.ylabel("frequency of $A$ allele, $p$")
plt.show()
</pre>

<p><span id='section3'></span></p>
<h2 id="3-one-locus-diploid-selection-in-discrete-time">3. One-locus diploid selection in discrete time</h2>
<hr>

<p>Since we are all currently in the diploid phase of our life-cycle, it is natural to ask: <em>Does selection in the diploid phase work the same way?</em></p>
<h3 id="life-cycle-diagram_1">Life-cycle diagram</h3>
<p>We now have the following life-cycle diagram.</p>
<p><center></p>
<pre class="mermaid"><code>    graph LR;
    A((random&lt;br&gt; union)) --diploid--&gt; B((selection));
    B --diploid--&gt; C((meiosis));
    C --haploid--&gt; A;</code></pre>
<p></center></p>
<p><center><sup>Figure. Life-cycle diagram of diploid selection</sup></center></p>
<p>We will census at the beginning of the diploid phase (immediately after gamete union).</p>
<h3 id="deriving-the-equations_1">Deriving the equations</h3>
<p>Let</p>
<ul>
<li><span class="arithmatex">\(n_{AA}(t) =\)</span> number of <span class="arithmatex">\(AA\)</span> genotypes in generation <span class="arithmatex">\(t\)</span></li>
<li><span class="arithmatex">\(n_{Aa}(t) =\)</span> number of <span class="arithmatex">\(Aa\)</span> genotypes in generation <span class="arithmatex">\(t\)</span></li>
<li><span class="arithmatex">\(n_{aa}(t) =\)</span> number of <span class="arithmatex">\(aa\)</span> genotypes in generation <span class="arithmatex">\(t\)</span></li>
</ul>
<p>The frequency of allele <span class="arithmatex">\(A\)</span> is then</p>
<div class="arithmatex">\[
\begin{aligned}
p_t
&amp;= \frac{2n_{AA}(t) + n_{Aa}(t)}{2n_{AA}(t) + n_{Aa}(t) + n_{aa}(t)} \\
&amp;= \frac{n_{AA}(t) + \frac{1}{2}n_{Aa}(t)}{n_{AA}(t) + n_{Aa}(t) + n_{aa}(t)}
\end{aligned}
\]</div>
<p>Now let's assume that not all individuals survive to reproduce, and that the viability of an individual depends on its genotype:</p>
<ul>
<li><span class="arithmatex">\(V_{AA} =\)</span> fraction of <span class="arithmatex">\(AA\)</span> genotypes that survive</li>
<li><span class="arithmatex">\(V_{Aa} =\)</span> fraction of <span class="arithmatex">\(Aa\)</span> genotypes that survive</li>
<li><span class="arithmatex">\(V_{aa} =\)</span> fraction of <span class="arithmatex">\(aa\)</span> genotypes that survive</li>
</ul>
<p>After these deaths there will be <span class="arithmatex">\(v_{AA}n_{AA}(t)\)</span> <span class="arithmatex">\(AA\)</span> genotypes, <span class="arithmatex">\(v_{Aa}n_{Aa}(t)\)</span> <span class="arithmatex">\(Aa\)</span> genotypes, and <span class="arithmatex">\(v_{aa}n_{aa}(t)\)</span> <span class="arithmatex">\(aa\)</span> genotypes. The frequency of the <span class="arithmatex">\(A\)</span> allele among these survivors is therefore</p>
<div class="arithmatex">\[
\frac{v_{AA}n_{AA}(t) + \frac{1}{2}v_{Aa}n_{Aa}(t)}{v_{AA}n_{AA}(t) + \frac{1}{2}v_{Aa}n_{Aa}(t) + v_{aa}n_{aa}(t)}.
\]</div>
<p>Since meiosis, passage through the haploid phase without selection, and random union will not change this frequency, this is the frequency of the <span class="arithmatex">\(A\)</span> allele in the next generation,</p>
<div class="arithmatex">\[
p_{t+1} = \frac{v_{AA}n_{AA}(t) + \frac{1}{2}v_{Aa}n_{Aa}(t)}{v_{AA}n_{AA}(t) + \frac{1}{2}v_{Aa}n_{Aa}(t) + v_{aa}n_{aa}(t)}.
\]</div>
<p>As above, we want a recursion equation, so we want to replace the <span class="arithmatex">\(n_i\)</span>'s in the right hand side of this equation with <span class="arithmatex">\(p\)</span>'s. To do this we note that with the random union of gametes the diploid offspring are in Hardy-Weinberg proportions, so that</p>
<ul>
<li><span class="arithmatex">\(n_{AA}(t) = p_t^2 n(t)\)</span></li>
<li><span class="arithmatex">\(n_{Aa}(t) = 2p_t q_t n(t)\)</span></li>
<li><span class="arithmatex">\(n_{aa}(t) = q_t^2 n(t)\)</span></li>
</ul>
<p>where <span class="arithmatex">\(n(t) = n_{AA}(t) + n_{Aa}(t) + n_{aa}(t)\)</span> is the total population size.</p>
<p>Substituting these Hardy-Weinberg proportions in and simplifying, the total population size cancels out and we can rewrite the above equation in terms of allele frequency alone,</p>
<div class="arithmatex">\[
p_{t+1} = \frac{v_{AA}p_t^2 + v_{Aa}p_t q_t}{v_{AA}p_t^2 + 2v_{Aa}p_t q_t + v_{aa}q_t^2}.
\]</div>
<p>This is a recursion equation for allele frequency in our model of diploid selection.</p>
<h3 id="simplifying_1">Simplifying</h3>
<p>As in the haploid selection case, we can divide by one of the viabilities, say <span class="arithmatex">\(v_{aa}\)</span>, to remove one parameter from the model</p>
<div class="arithmatex">\[
\begin{aligned}
p_{t+1}
&amp;= \frac{(v_{AA}/v_{aa})p_t^2 + (v_{Aa}/v_{aa})p_t q_t}{(v_{AA}/v_{aa})p_t^2 + 2(v_{Aa}/v_{aa})p_t q_t + (v_{aa}/v_{aa})q_t^2} \\
&amp;= \frac{W_{AA}p_t^2 + W_{Aa}p_t q_t}{W_{AA}p_t^2 + 2W_{Aa}p_t q_t + q_t^2}
\end{aligned}
\]</div>
<p>This recursion is a function of only two relative viabilities, <span class="arithmatex">\(W_{AA} = v_{AA}/v_{aa}\)</span> and <span class="arithmatex">\(W_{Aa} = v_{Aa}/v_{aa}\)</span>.</p>
<p>Let's plot the recursion to get a sense of the dynamics</p>
<pre data-executable="true" data-language="python">
import matplotlib.pyplot as plt

# calculate allele frequency over time with recursion
W_AA = 1.2 #relative viability of genotype AA
W_Aa = 1.1 #relative viability of genotype Aa 
p_now = 0.01 #initial allele frequency, p_0
ps = [] #list to hold allele frequencies over time
ts = range(100) #list of time steps
for t in ts: #for each time
    ps.append(p_now) #add current allele frequency to list
    p_now = (W_AA*p_now**2 + W_Aa*p_now*(1-p_now))/(W_AA*p_now**2 + 2*W_Aa*p_now*(1-p_now) + (1-p_now)**2) #update allele frequency with our recursion equation

# plot
plt.scatter(ts, ps)
plt.xlabel("generation, $t$")
plt.ylabel("frequency of $A$ allele, $p$")
plt.show()
</pre>

<h2 id="4-comparing-haploid-and-diploid-selection">4. Comparing haploid and diploid selection</h2>
<hr>

<p>So, returning to our original question, let's compare evolution under haploid selection</p>
<div class="arithmatex">\[
p_{t+1} = \frac{v_A p_t}{v_A p_t + v_a q_t}
\]</div>
<p>to evolution under diploid selection</p>
<div class="arithmatex">\[
p_{t+1} = \frac{v_{AA}p_t^2 + v_{Aa}p_t q_t}{v_{AA}p_t^2 + 2v_{Aa}p_t q_t + v_{aa}q_t^2}.
\]</div>
<p>To facilitate this, let’s assume the viability of a diploid genotype is the product of the haploid viabilities, i.e., <span class="arithmatex">\(v_{AA} = v_Av_A\)</span>, <span class="arithmatex">\(v_{Aa} = v_Av_a\)</span>, and <span class="arithmatex">\(v_{aa} = v_av_a\)</span>.</p>
<p>It then happens that our diploid recursion reduces to the haploid recursion,</p>
<div class="arithmatex">\[
\begin{aligned}
p_{t+1} &amp;= 
\frac{v_A v_A p_t^2 + v_A v_a p_t q_t}{v_A v_A p_t^2 + 2v_A v_a p_t q_t + v_a v_a q_t^2}\\
&amp;= \frac{v_Ap_t(p_t + v_a q_t)}{v_A p_t(v_A p_t + v_a q_t) + v_a q_t(v_A p_t + v_a q_t)}\\
&amp;= \frac{v_A p_t}{v_A p_t + v_a q_t}
\end{aligned}
\]</div>
<p>This shows that we need twice as much selection under diploid selection relative to that under haploid selection (e.g., <span class="arithmatex">\(v_{AA} = v_Av_A\)</span>) for evolution to proceed as quickly. <em>Why is evolution slower under diploid selection?</em></p>
<p>Let's check this visually (try breaking the assumption to see what happens)</p>
<pre data-executable="true" data-language="python">
import matplotlib.pyplot as plt

# calculate allele frequency over time with recursion
W_A = 1.1 #relative viability of allele A in the haploid model
W_AA = W_A*W_A #relative viability of genotype AA in the diploid model
W_Aa = W_A #relative viability of genotype Aa in the diploid model
p_now_hap = 0.01 #initial allele frequency, p_0
p_now_dip = p_now_hap
ps_hap, ps_dip = [], [] #lists to hold allele frequencies over time
ts = range(100) #list of time steps
for t in ts: #for each time
    ps_hap.append(p_now_hap) #add current allele frequency to list
    p_now_hap = W_A*p_now_hap/(W_A*p_now_hap+(1-p_now_hap)) #update allele frequency with haploid recursion equation
    ps_dip.append(p_now_dip) #add current allele frequency to list
    p_now_dip = (W_AA*p_now_dip**2 + W_Aa*p_now_dip*(1-p_now_dip))/(W_AA*p_now_dip**2 + 2*W_Aa*p_now_dip*(1-p_now_dip) + (1-p_now_dip)**2) #update allele frequency with diploid recursion equation

# plot
plt.scatter(ts, ps_hap, label='haploid')
plt.scatter(ts, ps_dip, label='diploid')
plt.xlabel("generation, $t$")
plt.ylabel("frequency of $A$ allele, $p$")
plt.legend() #add legend to know which points belong to which model
plt.show()
</pre>

<p><span id='section4'></span></p>
<h2 id="5-one-locus-haploid-selection-in-continuous-time">5. One-locus haploid selection in continuous time</h2>
<hr>

<p>To model haploid selection in continuous-time we assume each haploid genotype has growth rate</p>
<ul>
<li><span class="arithmatex">\(r_A =\)</span> growth rate of the <span class="arithmatex">\(A\)</span> genotype</li>
<li><span class="arithmatex">\(r_a =\)</span> growth rate of the <span class="arithmatex">\(a\)</span> genotype</li>
</ul>
<p>which takes into account the time spent going through gamete union, diploidy, and meiosis.</p>
<p>We therefore have exponential growth of both genotypes according to</p>
<div class="arithmatex">\[\frac{\mathrm{d} n_A}{dt} = r_A n_A\]</div>
<div class="arithmatex">\[\frac{\mathrm{d} n_a}{dt} = r_a n_a.\]</div>
<p>At any particular point in time, <span class="arithmatex">\(t\)</span>, the frequency of allele <span class="arithmatex">\(A\)</span> is as above, <span class="arithmatex">\(p_t = n_A(t)/(n_A(t) + n_a(t))\)</span>. </p>
<p>We can therefore derive the rate of change in the frequency of allele <span class="arithmatex">\(A\)</span>, <span class="arithmatex">\(\mathrm{d}p/\mathrm{d}t\)</span>, using the qoutient rule</p>
<div class="arithmatex">\[
\begin{aligned}
\frac{\mathrm{d}p}{\mathrm{d}t}
&amp;= \frac{\mathrm{d}\frac{n_A}{n_A + n_a}}{\mathrm{d}t} \\
&amp;= \frac{\frac{\mathrm{d}n_A}{\mathrm{d}t} (n_A + n_a) - n_A \frac{\mathrm{d}(n_A+n_a)}{\mathrm{d}t}}{(n_A + n_a)^2}\\
&amp;= \frac{r_A n_A (n_A + n_a) - n_A (r_A n_A + r_a n_a)}{(n_A + n_a)^2}\\
&amp;= \frac{r_A n_A n_a - r_a n_A n_a)}{(n_A + n_a)^2}\\
&amp;= \frac{(r_A -r_a)n_A n_a}{(n_A + n_a)^2}\\
&amp;= (r_A -r_a)p q,
\end{aligned}
\]</div>
<p>where <span class="arithmatex">\(s_c = r_A - r_a\)</span> is the continuous-time <strong>selection coefficient</strong> of allele <span class="arithmatex">\(A\)</span>.</p>
<p>A similar equation can be derived for the model of diploid-selection in continuous time, but we will not study it (see the textbook if you are curious).</p>
<p><span id='section6'></span></p>
<h2 id="6-comparing-haploid-selection-in-discrete-and-continuous-time">6. Comparing haploid selection in discrete and continuous time</h2>
<hr>

<p>Are the discrete- and continuous-time models of haploid selection as different as they look?</p>
<div class="arithmatex">\[
p_{t+1} = \frac{v_A p_t}{v_A p_t + v_a q_t}
\]</div>
<div class="arithmatex">\[
\frac{\mathrm{d}p}{\mathrm{d}t} = s_c p q
\]</div>
<p>Not really. Discrete and continuous time models generally behave in a similar fashion when changes occur slowly over time. For this model of haploid selection, this implies that the discrete and continuous models will be similar when the fitnesses of the two genotypes are nearly equal, i.e., when <span class="arithmatex">\(v_A - v_a\)</span> is small. This is called "weak selection".</p>
<p>In the discrete model, the change in the allele frequency is</p>
<div class="arithmatex">\[
\begin{aligned}
\Delta p &amp;= p_{t+1} - p_t \\
&amp;= \frac{v_A p_t}{v_A p_t + v_a q_t} - p_t \\
&amp;= \frac{(v_A - v_a)p_t q_t}{v_A p_t + v_a q_t}.
\end{aligned}
\]</div>
<p>Now define <span class="arithmatex">\(s_d = (v_A - v_a)/v_a\)</span> as the discrete time selection coefficient. Subbing this in as <span class="arithmatex">\(V_A = s_d v_a + v_a\)</span> gives</p>
<div class="arithmatex">\[
\begin{aligned}
\Delta p 
&amp;= \frac{(v_A-v_a)p_t q_t}{v_A p_t + v_a q_t}\\
&amp;= \frac{(s_d v_a + v_a - v_a) p_t q_t}{(s_d v_a + v_a) p_t + v_a q_t}\\
&amp;= \frac{s_d v_a p_t q_t}{s_d v_a p_t + v_a (p_t + q_t)}\\
&amp;= \frac{s_d v_a p_t q_t}{s_d v_a p_t + v_a}\\
&amp;= \frac{s_d p_t q_t}{s_d p_t + 1}\\
\end{aligned}
\]</div>
<p>Now assume weak selection, ie, that <span class="arithmatex">\(s_d\)</span> is small. This implies <span class="arithmatex">\(s_d p_t + 1 \approx 1\)</span>. Making this approximation we have</p>
<div class="arithmatex">\[
\Delta p \approx s_d p_t q_t
\]</div>
<p>This is equivalent to the continuous-time model when the selection coefficients are equal, <span class="arithmatex">\(s_c = s_d\)</span>.</p>
<p><span id='section7'></span></p>
<h2 id="7-the-models-weve-covered">7. The models we've covered</h2>
<hr>

<p>To summarize the last two lectures, we've derived four of the most classic models in ecology and evolution:</p>
<table>
<thead>
<tr>
<th>Model</th>
<th>Discrete time</th>
<th>Continous time</th>
</tr>
</thead>
<tbody>
<tr>
<td>Exponential growth</td>
<td><span class="arithmatex">\(n_{t+1} = R n_t\)</span></td>
<td><span class="arithmatex">\(\frac{\mathrm{d}n}{\mathrm{d}t} = r n\)</span></td>
</tr>
<tr>
<td>Logistic growth</td>
<td><span class="arithmatex">\(n_{t+1} = (1 + r(1 - \frac{n_t}{K}))n_t\)</span></td>
<td><span class="arithmatex">\(\frac{\mathrm{d}n}{\mathrm{d}t} = r(1 - \frac{n}{K})n\)</span></td>
</tr>
<tr>
<td>Haploid selection</td>
<td><span class="arithmatex">\(p_{t+1} = \frac{p_t W_A}{p_t W_A + q_t}\)</span></td>
<td><span class="arithmatex">\(\frac{\mathrm{d}p}{\mathrm{d}t} = s p(1-p)\)</span></td>
</tr>
<tr>
<td>Diploid selection</td>
<td><span class="arithmatex">\(p_{t+1} = \frac{p_t^2 W_{AA} + p_t q_t W_{Aa}}{p_t^2 W_{AA} + 2 p_t q_t W_{Aa} + q_t^2}\)</span></td>
<td>Not derived</td>
</tr>
</tbody>
</table>
<p>See textbook sections 3.4 and 3.5 for models of interacting species and epidemiology, respectively, which we won't cover in class.</p>




              
            </article>
            
          </div>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
    
    <nav class="md-footer__inner md-grid" aria-label="Footer" >
      
        
        <a href="../lecture-03/" class="md-footer__link md-footer__link--prev" aria-label="Previous: Lecture 3" rel="prev">
          <div class="md-footer__button md-icon">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
          </div>
          <div class="md-footer__title">
            <div class="md-ellipsis">
              <span class="md-footer__direction">
                Previous
              </span>
              Lecture 3
            </div>
          </div>
        </a>
      
      
        
        <a href="../../labs/schedule/" class="md-footer__link md-footer__link--next" aria-label="Next: Schedule" rel="next">
          <div class="md-footer__title">
            <div class="md-ellipsis">
              <span class="md-footer__direction">
                Next
              </span>
              Schedule
            </div>
          </div>
          <div class="md-footer__button md-icon">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11H4Z"/></svg>
          </div>
        </a>
      
    </nav>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    <script id="__config" type="application/json">{"base": "../..", "features": ["content.code.annotate", "navigation.tabs"], "search": "../../assets/javascripts/workers/search.ecf98df9.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.config.lang": "en", "search.config.pipeline": "trimmer, stopWordFilter", "search.config.separator": "[\\s\\-]+", "search.placeholder": "Search", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version.title": "Select version"}}</script>
    
    
      <script src="../../assets/javascripts/bundle.9c69f0bc.min.js"></script>
      
        <script src="../../javascripts/mathjax.js"></script>
      
        <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
      
        <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
      
    
  </body>
</html>